simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/Evse.java

```diff
@@ -250,6 +232,95 @@ public void clearToken() {
         tokenId = StringUtils.EMPTY;
     }
 
+
+    /**
+     * Try to plug connector. If EVSE is unavailable it returns false.
+     *
+     * @param connectorId connector identity
+     * @return true if succeeded otherwise false
+     */
+    public boolean tryPlug(Integer connectorId) {
+
+        if (evseStatus.isUnAvailable()) {
+            return false;
+        }
+
+        Connector connector = findConnector(connectorId);
+        connector.plug();
+
+        return true;
+    }
+
+    /**
+     * Try to unplug connector. if EVSE is unavailable method returns false.
+     *
+     * @param connectorId connector identity
+     * @return true if succeeded otherwise false
+     */
+    public boolean tryUnplug(Integer connectorId) {
+
+        if (evseStatus.isUnAvailable()) {
+            return false;
+        }
+
+        Connector connector = findConnector(connectorId);
+        connector.unplug();
+
+        return true;
+    }
+
+
+    /**
+     * Find any PLUGGED connector and switch to LOCKED status.
+     *
+     * @return identity of the connector.
+     */
+    public Integer lockPluggedConnector() {
+
+        if (evseStatus.isUnAvailable()) {
+            throw new IllegalStateException("Can not lock connector when EVSE status is UnAvailable");
+        }
+
+        Connector pluggedConnector = connectors.stream()
+                .filter(Connector::isCablePlugged)
+                .findAny()
+                .orElseThrow(() -> new IllegalStateException(String.format("Unable to lock connector (nothing is plugged in): evseId=%s", id)));
+
+        pluggedConnector.lock();
+
+        return pluggedConnector.getId();
+    }
+
+    /**
+     * Find any LOCKED connector and switch to PLUGGED status.
+     *
+     * @return identity of the connector.
+     */
+    public Integer unlockConnector() {
+
+        if (evseStatus.isUnAvailable()) {
```

Comment:
isUnavailable?

Reply:
Are you sure that connector should stay locked if evse is unavailable?
I wouldn't add this condition for now.
OR can we check evse availability when token authorization is performed?
if (evse.hasOngoingTransaction()) {
//allow auth to unlock cable
} else {
//authorization is not allowed
}
Perhaps user won't be able to authorize if evse is disabled, but if there's an ongoing transaction, then even in unavailable state it could handle last authorization to unlock the cable, although I'm not sure what spec says about this case.

Reply:
Removed that.